CMakeLists.txt

```
@@ -12,11 +12,14 @@ if(NOT WIN32)
       ncursesw5-config --terminfo-dirs 2>/dev/null || \
       ncurses5-config  --terminfo-dirs 2>/dev/null || \
       echo '/etc/terminfo:/lib/terminfo:/usr/share/terminfo:/usr/lib/terminfo:/usr/local/share/terminfo:/usr/local/lib/terminfo'"
-    OUTPUT_VARIABLE TERMINFO_DIRS)
+    OUTPUT_VARIABLE TERMINFO_DIRS_DEFAULT)
   # Remove trailing newline
-  string(STRIP "${TERMINFO_DIRS}" TERMINFO_DIRS)
+  string(STRIP "${TERMINFO_DIRS_DEFAULT}" TERMINFO_DIRS_DEFAULT)
 else()
-  set(TERMINFO_DIRS "")
+  set(TERMINFO_DIRS_DEFAULT "")
+endif()
+if(NOT DEFINED TERMINFO_DIRS)
+  set(TERMINFO_DIRS "${TERMINFO_DIRS_DEFAULT}")
 endif()
```

nit: I think I'd prefer a cache variable if that works, mostly for discoverability when using `ccache` (although it's pretty small at this point but you never know :) ).

```suggestion
set(TERMINFO_DIRS "${TERMINFO_DIRS_DEFAULT}" CACHE PATH "<Insert appropriate comment IDK what TERMINFO_DIRS is>")
```

---

I was going to do this, but I was under the impression doing this depended on a minimum version of CMake, except I don't know what version that is...

---

And by `ccache`, I assume you mean `ccmake`?

---

Applied, with minor revisions.